### ISSUE TITLE
Bad example in Applied Accessibility/Use tabindex to Add Keyboard Focus to an Element

### ISSUE THREAD

**Issue by bbsmooth** [freeCodeCamp/freeCodeCamp#42311]
**Affected page**
https://www.freecodecamp.org/learn/responsive-web-design/applied-accessibility/use-tabindex-to-add-keyboard-focus-to-an-element

**Issue**
The instructions in this lesson are:

"He knows input fields automatically get keyboard focus, but **he wants to make sure his keyboard users pause at the instructions while tabbing through the items**. Add a tabindex attribute to the p tag and set its value to 0."

This is a big accessibility anti-pattern. In general, tabindex="0" should only be set on elements that are interactive. There are other more accessible options for associating the instructions with the form. First, you would want to move the instructions just before the opening form element to ensure they are read before a screen reader goes into forms mode. You might even associate the instructions with the first input field in the form using aria-describedby but I think a better method would be to indicate which inputs are required individually using one of several options I won't go into here.

Regardless of how you ultimately handle this issue, the one way you do not want to do it is to add tabindex="0" to the p tag in order to insert that paragraph into the natural tab order. 

**References**
https://www.w3.org/TR/html53/editing.html#the-tabindex-attribute
https://www.w3.org/WAI/tutorials/forms/instructions/
https://www.a11yproject.com/posts/2021-01-28-how-to-use-the-tabindex-attribute/

**Suggested Alternative Example**
Generally you would use tabindex="0" when creating custom interactive components, which might be too much to explain for this lesson? I suppose you could use a simpler example that turns a div into a link (or something similar) but then not only do you need to add tabindex but you also need to add a role and explain that you have to use javascript to implement all of the functionality that a link would normally have and also explain that most accessibility experts would recommend you always use an actual anchor tag instead of trying to turn a div into something it's not.

One example that does come to mind is setting tabindex="0" on an element that will scroll when it overflows. Some browsers don't automatically add these elements to the tab order and thus keyboard users can't scroll them without the added tabindex. For an example see https://adrianroselli.com/2020/11/under-engineered-responsive-tables.html.

Another option might be to use tabindex="-1". This is often used to send focus to a non-interactive element. For example, we use it for quizzing, when the student submits their answer we remove the submit button and set the focus on the element that contains the feedback (which has tabindex="-1" set on it).

**Comment by moT01** [comment]
I think the purpose of this lesson is to show how to give a tabindex to elements that don't have them by default. So it sounds like changing the lesson to use a -1 tabindex instead of 0 is the way to go. That should be an easy fix as it shouldn't require changing any (or very little) of the instructions. How does that sound @bbsmooth? And any interest in making a PR?

**Comment by hchiam** [comment]
Hi, first time here! I made a PR for this: https://github.com/freeCodeCamp/freeCodeCamp/pull/42578

**Comment by ShaunSHamilton** [comment]
Re https://github.com/freeCodeCamp/freeCodeCamp/pull/42578#pullrequestreview-690462494

My suggestion is to add a small pop-up/modal, when the user clicks the submit button, which can receive the `tabIndex=-1`. It could even be something as simple as a hidden `p` element.

**Comment by bbsmooth** [comment]
Sorry, it's been a while since I've had a chance to get back to this. Another suggestion for updating this challenge would be for the user to fix an inappropriate usage of `tabindex="0"` by having them remove it from the HTML. The challenge currently explains what `tabindex="0"` does to an element. It could then elaborate that this should only be used in very specific circumstances, such as for custom widgets, and should only be used on an element that has some sort functionality associated with it.

The HTML could start with the `tabindex="0"` on the `<p>`. The instructions could explain that the developer thought that adding this would help screen reader users by forcing them to tab through the instructions but in reality this is not a proper use of tabindex because the `<p>` is not an interactive element and there are more appropriate ways to help ensure screen reader users hear the instructions. The user passes the challenge once they have removed the tabindex from the `<p>`.
